File: flink-code-java-0421/src/main/java/com/atguigu/day02/UnionExample.java
package com.atguigu.day02;

import com.atguigu.day02.util.SensorReading;
import com.atguigu.day02.util.SensorSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class UnionExample {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        SingleOutputStreamOperator<SensorReading> sensor_1 = env.addSource(new SensorSource()).filter(r -> r.id.equals("sensor_1"));
        SingleOutputStreamOperator<SensorReading> sensor_2 = env.addSource(new SensorSource()).filter(r -> r.id.equals("sensor_2"));
        SingleOutputStreamOperator<SensorReading> sensor_3 = env.addSource(new SensorSource()).filter(r -> r.id.equals("sensor_3"));

        sensor_1.union(sensor_2, sensor_3).print();

        env.execute();
    }
}

File: Flink0105/src/main/scala/com/atguigu/day2/UnionExample.scala
package com.atguigu.day2

import org.apache.flink.streaming.api.scala._

object UnionExample {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)

    val BJ : DataStream[SensorReading] = env
      .addSource(new SensorSource)
      .filter(_.id.equals("sensor_1"))

    val SH : DataStream[SensorReading] = env
      .addSource(new SensorSource)
      .filter(_.id.equals("sensor_2"))

    val SZ : DataStream[SensorReading] = env
      .addSource(new SensorSource)
      .filter(_.id.equals("sensor_3"))

    val union : DataStream[SensorReading] = BJ.union(SH, SZ)

    union.print()

    env.execute()
  }
}